src/main/java/com/google/devtools/build/lib/rules/cpp/CcToolchainVariables.java

```diff
@@ -784,11 +806,17 @@ public VariableValue getFieldValue(
           String variableName,
           String field,
           @Nullable ArtifactExpander expander,
+          PathMapper pathMapper,
           boolean throwOnMissingVariable) {
         if (NAME_FIELD_NAME.equals(field)) {
-          return new StringValue(name);
+          if (pathMapper.isNoop()) {
+            return new StringValue(name);
+          }
```

Comment:
Is this an optimization or would the no-op PathMapper produce wrong result executing next statment?

Reply:
Assuming that the `name` field only contains normalized file paths (which I guess it does?), this meant to be a pure optimization.